app/services/work_files_scanner.rb

```diff
@@ -0,0 +1,56 @@
+class WorkFilesScanner
+
+  attr_reader :ability, :results, :work_id
+
+  delegate :file_count, :file_size_total, to: :results
+
+  ScanResults = Struct.new(:file_count, :file_size_total)
+
+  def self.call(work_id, ability=nil)
+    self.new(work_id, ability).scan
+  end
+
+  def initialize(work_id, ability)
+    @work_id = work_id
+    @ability = ability
+  end
+
+  def scan
+    init_scan_results
+    work_doc = SolrDocument.find(work_id)
+    scan_work(work_doc)
+    results
+  end
+
+  private
+
+  def init_scan_results
+    @results = ScanResults.new.tap do |r|
+      r.file_count = 0
+      r.file_size_total = 0
+    end
+  end
+
+  def scan_work(work_doc)
+    work_doc.members.each do |member|
+      member_doc = SolrDocument.find(member)
+      if permitted?(member_doc)
+        if member_doc.hydra_model == FileSet
+          scan_fileset(member_doc)
+        else
+          scan_work(member_doc)
+        end
+      end
+    end
+  end
+
+  def scan_fileset(fileset_doc)
+    results.file_count = results.file_count + 1
+    results.file_size_total = results.file_size_total + fileset_doc['file_size_lts']
```

Comment:
I'm getting a `nil can't be coerced into Fixnum` error here locally for some works. Not quite sure what conditions trigger though. Possibly a mix of child works with files & no files? Parent works having both files and children works? I'll try to track that down.

Reply:
I see the same error, but only for one item in my local instance. It's got three files in it - not sure why it's choking when other local items are not...

Reply:
I'll take a look and see if I can replicate.  Thanks.